File: VB/FindPanelExample/MainForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports System.Collections.Generic
Imports DevExpress.XtraGrid.Views.Grid

Namespace FindPanelExample
	Partial Public Class MainForm
		Inherits XtraForm
		Private Const Alphabet As String = "abcdefghigklmnopqarstuvwxyz"
		Private Shared Randomizer As New Random()

		Public Sub New()
			InitializeComponent()
			GridControl.BeginUpdate()
			Try
				For i As Integer = 0 To 999
					For j As Integer = 0 To 9
						dataTable1.Rows.Add(GetRandomText(), i)
					Next j
				Next i
			Finally
				GridControl.EndUpdate()
			End Try
		End Sub

		Private Shared Function GetRandomText() As String
			Dim result As New StringBuilder()
			For i As Integer = 0 To 9
				result.Append(Alphabet.Chars(Randomizer.Next(27)))
			Next i
			Return result.ToString()
		End Function

		Private Sub OnGridViewRowCountChanged(ByVal sender As Object, ByVal e As EventArgs) Handles GridView.RowCountChanged
			If String.IsNullOrEmpty(GridView.FindFilterText) Then
				Return
			End If
			BeginInvoke(New MethodInvoker(AddressOf ExpandVisibleRows))
		End Sub

		Private Sub ExpandVisibleRows()
            Dim i As Integer = GridView.TopRowIndex
            Dim rowHandle As Integer = GridView.GetVisibleRowHandle(i)
            Do While GridView.IsRowVisible(rowHandle) = RowVisibleState.Visible
                If GridView.IsGroupRow(rowHandle) Then
                    GridView.ExpandGroupRow(rowHandle)
                End If
                i += 1
                rowHandle = GridView.GetVisibleRowHandle(i)
            Loop
		End Sub
	End Class
End Namespace

File: CS/FindPanelExample/MainForm.cs
using System;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using System.Collections.Generic;
using DevExpress.XtraGrid.Views.Grid;

namespace FindPanelExample {
    public partial class MainForm :XtraForm {
        const string Alphabet = "abcdefghigklmnopqarstuvwxyz";
        static Random Randomizer = new Random();

        public MainForm() {
            InitializeComponent();
            GridControl.BeginUpdate();
            try {
                for (int i = 0; i < 1000; i++)
                    for (int j = 0; j < 10; j++)
                        dataTable1.Rows.Add(GetRandomText(), i);
            } finally { GridControl.EndUpdate(); }
        }

        static string GetRandomText() {
            StringBuilder result = new StringBuilder();
            for (int i = 0; i < 10; i++)
                result.Append(Alphabet[Randomizer.Next(27)]);
            return result.ToString();
        }

        void OnGridViewRowCountChanged(object sender, EventArgs e) {
            if (string.IsNullOrEmpty(GridView.FindFilterText)) return;
            BeginInvoke(new MethodInvoker(ExpandVisibleRows));
        }

        void ExpandVisibleRows() {
            int rowHandle;
            for (int i = GridView.TopRowIndex; GridView.IsRowVisible(rowHandle =
                GridView.GetVisibleRowHandle(i)) == RowVisibleState.Visible; i++)
                if (GridView.IsGroupRow(rowHandle))
                    GridView.ExpandGroupRow(rowHandle);
        }
    }
}